### PR TITLE
feat(metagen): `mdk_rs` finalization

### PR DESCRIPTION
**Pull request by Yohe-Am**
- Finalizes the host side of the `wit_wire` implementation started in #687 .
- Fixes a few issues with the `mdk_rust` generator.

#### Motivation and context

The code generator was already in place but we the typegate had no support for the `wit_wire` interface used by the rust mdk. This PR adds that.

The two week delay is mainly due to the base work required in #669 and related PRs.

#### Migration notes

_No breaking changes on user._

### Checklist

- [x] The change come with new or modified tests
- [x] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

### CODE REVIEW

**Review by afmika — COMMENTED** [review]
🔥

File: typegate/src/runtimes/wasm.ts [hunk 1]
```diff
@@ -2,107 +2,21 @@
...
+import { RuntimeInitParams } from "../types.ts";
 import { registerRuntime } from "./mod.ts";
-import * as ast from "graphql/ast";
-import { Materializer } from "../typegraph/types.ts";
-import { Typegate } from "../typegate/mod.ts";
+import { WasmRuntimeData } from "../typegraph/types.ts";
+import { WasmRuntimeReflected } from "./wasm/reflected.ts";
+import { WasmRuntimeWire } from "./wasm/wire.ts";
 
 @registerRuntime("wasm")
```
Comment (Yohe-Am) [inline typegate/src/runtimes/wasm.ts:11]: Not here, I've mapped both wire and reflected wasm runtimes to `wasm` as their runtime name. Disambiguation is done on the runtime data instead. I'm unsure if this is the a right decision and am willing to refactor to distinct names.

File: libs/metagen/src/mdk_rust/static/lib.rs [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
+// SPDX-License-Identifier: MPL-2.0
```
Comment (zifeo) [inline libs/metagen/src/mdk_rust/static/lib.rs:2]: is this file required?

File: libs/metagen/tests/mat_rust/mdk.rs [hunk 3]
```diff
@@ -0,0 +1,2 @@
+// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
+// SPDX-License-Identifier: MPL-2.0
```
Comment (zifeo) [inline libs/metagen/tests/mat_rust/mdk.rs:2]: also here?

File: typegate/tests/runtimes/python/python_test.ts [hunk 4]
```diff
@@ -141,7 +140,7 @@ Meta.test(
         .on(e);
     });
 
-    await t.should("work once (module)", async () => {
+    /* await t.should("work once (module)", async () => {
```
Comment (zifeo) [inline typegate/tests/runtimes/python/python_test.ts:143]: a leftover to uncomment?

File: typegate/tests/runtimes/wasm_wire/rust.wasm
Comment (zifeo) [inline typegate/tests/runtimes/wasm_wire/rust.wasm:1]: can we wasm-opt that one to get a lighter file or even remove the debug symbols?

File: ghjk.ts [hunk 5]
```diff
@@ -144,9 +145,10 @@ ghjk.task("build-pyrt", {
   allowedPortDeps,
   dependsOn: ["gen-pyrt-bind"],
   async fn({ $, argv, env }) {
-    const target = env["PYRT_TARGET"] ? `--target ${env["PYRT_TARGET"]}` : "";
     const wasmOut = env["PYRT_WASM_OUT"] ?? "./target/pyrt.wasm";
-    const cwasmOut = env["PYRT_CWASM_OUT"] ?? "./target/pyrt.cwasm";
-    await $`componentize-py -d ./libs/pyrt_wit_wire/wit/ componentize -o ${wasmOut} libs.pyrt_wit_wire.main`;
+    // TODO: support for `world-module` is missing on the `componentize` subcmd
```
[the remark this answers is not in the document]
Comment (zifeo) [reply ghjk.ts:149]: I would vote for using 2 different names, but let's see other comments...

**Review by zifeo — COMMENTED** [review]
🚀 (needs a second-longer look but already some feedbacks)

File: typegate/tests/runtimes/wasm_wire/rust.wasm
Comment (Yohe-Am) [reply typegate/tests/runtimes/wasm_wire/rust.wasm:1]: Nice, shaved of 200KiB.

Comment (Yohe-Am) [reply typegate/tests/runtimes/wasm_wire/rust.wasm:1]: Oh wow, strip + opt-level = "z" got it down to less than 200KiBs even.

File: typegate/tests/runtimes/python/python_test.ts (hunk 4 above)
Comment (Yohe-Am) [reply typegate/tests/runtimes/python/python_test.ts:143]: This has been handled in #699.

File: libs/metagen/tests/mat_rust/mdk.rs (hunk 3 above)
Comment (Yohe-Am) [reply libs/metagen/tests/mat_rust/mdk.rs]: This one's not necessary indeed.

File: libs/metagen/src/mdk_rust/static/lib.rs (hunk 2 above)
Comment (Yohe-Am) [reply libs/metagen/src/mdk_rust/static/lib.rs:2]: Huh, that one's not supposed to be empty. Thanks.
